tests/system/utils.py

```diff
@@ -39,3 +40,122 @@ def get_resource(resource):
                 raise Exception
         except Exception:
             raise DCOSException("Can't read from resource: {0}. Please check that it exists.".format(resource))
+
+
+class FaultDomain:
+    """
+    High-level representation of a fault domain
+    """
+    def __init__(self, config):
+        # Make sure config is a dict
+        if not type(config) is dict:
+            config = {}
+
+        # Extract fault domain
+        fault_domain = config.get('fault_domain', {})
+
+        # Extract zone/region
+        self.zone = fault_domain.get('zone', {}).get('name', 'default')
+        self.region = fault_domain.get('region', {}).get('name', 'default')
+
+
+def get_cluster_local_domain():
```

Comment:
What do you think about passing the cluster state instead?

Reply:
All the other `*_cluste_*` functions in the `utils.py` are not using this approach. What's your reasoning for breaking the interface with this one?

Reply:
You mean the cluster leading master's state? Or is there a "cluster state" somewhere that I am not aware of? :thinking: 

This function is too small, and the 50% of it is just to fetch the master state. Therefore I don't think it makes sense to fragment it even further. But that's just my personal opinion.